politeiad/backendv2/tstorebe/plugins/comments/cmds.go

```diff
@@ -642,8 +705,17 @@ func (p *commentsPlugin) cmdEdit(token []byte, payload string) (string, error) {
 		}
 	}
 
+	// Verify extra data hint
+	if e.ExtraDataHint != existing.ExtraDataHint {
```

Comment:
Do you think this should be left up to the application plugin (i.e. the pi plugin) to decide?

Reply:
Hrmmm, do you mean adding a bool flag to determine whether changing the extra data hint 
is allowed or not ?

Reply:
do you mean adding something like the following for the comment edit ? not sure if the pi plugin would have 
access to the existing comment: `existing.ExtraDataHint`.

```
// hookCommentNew adds pi specific validation onto the comments plugin New
// command.
func (p *piPlugin) hookCommentNew(token []byte, cmd, payload string) error {
	return p.commentWritesAllowed(token, cmd, payload)
}
```

Reply:
The extra data fields are application specifc.  The comments plugin doesn't have any knowledge of what is a valid ExtraDataHint and what is an invalid ExtraDataHint.  This code makes the assumption that changing the extra data fields is not allowed, but that's not really up to the comments plugin to decide since those fields contain application data.  It should be up to the application plugin to decide.

Do you agree?

Reply:
I downside is the performance hit that you mentioned.  The pi plugin would need to query tstore for the comment, which would result in a tlog tree retrieval.  I think that for now, the way you have it is fine, but I would add in a comment explaining that this doesn't really belong here and is a result of u

Long term, we're going to need to do something about these tlog tree retrievals.  I'm thinking something like, lock the record while a write command is being executed and cache the tree results for the duration of the command.  This would allow the various plugin hooks to pulll data without causing a tlog tree retrieval.  We can figure that out later though.